LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/ClientCertificateValidatorService.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.NetworkServer.BasicsStation
+{
+    using System;
+    using System.Linq;
+    using System.Net.Security;
+    using System.Security.Cryptography.X509Certificates;
+    using System.Text.RegularExpressions;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Extensions.Logging;
+
+    internal class ClientCertificateValidatorService
+    {
+        private readonly IBasicsStationConfigurationService stationConfigurationService;
+        private readonly ILogger<ClientCertificateValidatorService> logger;
+
+        public ClientCertificateValidatorService(IBasicsStationConfigurationService stationConfigurationService,
+                                                 ILogger<ClientCertificateValidatorService> logger)
+        {
+            this.stationConfigurationService = stationConfigurationService;
+            this.logger = logger;
+        }
+
+        // Following synchronous implementation is needed for ASP.NET Core ClientCertificateValidation
+        public virtual bool Validate(X509Certificate2 certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors) =>
+            ValidateAsync(certificate, chain, sslPolicyErrors, default).GetAwaiter().GetResult();
+
+        public virtual async Task<bool> ValidateAsync(X509Certificate2 certificate, X509Chain? chain, SslPolicyErrors sslPolicyErrors, CancellationToken token)
+        {
+            if (certificate is null) throw new ArgumentNullException(nameof(certificate));
+            if (chain is null) throw new ArgumentNullException(nameof(chain));
+
+            var commonName = certificate.GetNameInfo(X509NameType.SimpleName, false);
+            StationEui stationEui;
+            if (Regex.Match(commonName, "([a-fA-F0-9]{2}[-:]?){8}") is { } regex && regex.Success)
+            {
+                stationEui = StationEui.Parse(regex.Value);
```

Comment:
It would be nice to handle this exception here (or use `TryParse` instead) and go through the `else` path as well with the same logging message. We can already anticipate that an exception might be thrown here if the EUI is not valid.